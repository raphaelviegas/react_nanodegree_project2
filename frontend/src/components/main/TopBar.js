import React from 'react'
import { Link } from 'react-router-dom'

export default () => {
    return (
        <div className="navbar-fixed">
            <nav>
                <div className="nav-wrapper cyan darken-3">
                    <span className="brand-logo"><Link to="/">Leituras</Link></span>
                </div>
            </nav>
        </div>
    )
}
