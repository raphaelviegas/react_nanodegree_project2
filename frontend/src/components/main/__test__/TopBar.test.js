import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router'

import TopBar from '../TopBar'
import { Link } from 'react-router-dom'

describe('<TopBar />', () => {
    let wrapper
    beforeEach(() => {
        wrapper = shallow(<TopBar />)
    })

    it('Should render one nav element', () => {
        expect(wrapper.find('nav').length).toEqual(1)
    })

    it('Should render one link tag to /', () => {
        expect(wrapper.containsMatchingElement(<Link to="/">Leituras</Link>)).toEqual(true)
    })

    it('Should render correctly', () => {
        const tree = renderer.create(
            <MemoryRouter initialEntries={[{ pathname: '/', key: 'testKey' }]}>
                <TopBar />
            </MemoryRouter>)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })

})
