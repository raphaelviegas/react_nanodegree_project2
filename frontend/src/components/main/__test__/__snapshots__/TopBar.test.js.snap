// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopBar /> Should render correctly 1`] = `
<div
  className="navbar-fixed"
>
  <nav>
    <div
      className="nav-wrapper cyan darken-3"
    >
      <span
        className="brand-logo"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Leituras
        </a>
      </span>
    </div>
  </nav>
</div>
`;
